device/cuda/src/clusterization/clusterization_algorithm.cu

```diff
@@ -310,11 +345,11 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
         cudaMemset(num_measurements_device.get(), 0, sizeof(unsigned int)));
 
     // Launch ccl kernel. Each thread will handle a single cell.
-    kernels::
-        ccl_kernel<<<num_partitions, m_max_cells_per_partition,
-                     2 * m_max_cells_per_partition * sizeof(index_t), stream>>>(
-            cells, modules, partitions, m_max_cells_per_partition,
-            measurements_buffer, *num_measurements_device);
+    kernels::ccl_kernel<<<
+        num_partitions, m_max_cells_per_partition / CELLS_PER_THREAD,
```

Comment:
Please make sure that this works if `m_max_cells_per_partition` is not a multiple of `CELLS_PER_THREAD`.

Reply:
that's a good idea